fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/rdf/impl/SkolemNodeRdfContext.java

```diff
@@ -93,6 +94,10 @@ public SkolemNodeRdfContext(final FedoraResource resource,
             return null;
 
         } catch (final RepositoryException e) {
+            if (e instanceof ItemNotFoundException) {
+                // Effectively skip resources that do not exist
```

Comment:
@birkland, the downside of potential, "invisible", dangling references (not resources) is not obvious. The only scenario that I can think of is if there is a single-valued WEAKREFERENCE property that is in the state of referencing a non-existent resource. Note, I say "WEAKREFERENCE" type because strong "REFERENCE" type already prevents such a scenario. I have tested this on the one REFERENCE type in Fedora's CND:
https://github.com/fcrepo4/fcrepo4/blob/master/fcrepo-kernel-modeshape/src/main/resources/fedora-node-types.cnd#L65

When I create a resource in the following situation:
`</a> ldp:membershipResource </x/y>`
..and attempt to delete </x>, a referential integrity exception is thrown. That means that this issue only relates to WEAKREFERENCEs, and since the Fedora CND does not define any WEAKREFERENCE types (moreless single-valued ones), I do not see a situation where invisible, dangling references will be a problem.

That said, a utility along the lines of [TechnicalMetadataMigrator](https://github.com/fcrepo4-exts/fcrepo4-upgrade-utils/blob/master/src/main/java/org/fcrepo/upgrade/utils/TechnicalMetadataMigrator.java#L51) would be useful as a "clean-up" tool.
I have created a ticket for implementing such a utility (noting that at the present, this ticket is a lower priority).
https://jira.duraspace.org/browse/FCREPO-2331

Additionally, I have pushed another commit with more logging.